src/components/nabla/Swap/From.tsx

```diff
@@ -54,15 +54,24 @@ export function From<FormFieldValues extends FieldValues, TFieldName extends Fie
           type="button"
         >
           <span className="mr-1 h-full rounded-full bg-[rgba(0,0,0,0.15)] p-px">
-            <img src={getIcon(fromToken?.symbol)} alt={fromToken?.name} className="h-full w-auto" />
+            <img src={getIcon(fromToken?.symbol)} alt={fromToken?.name} className="w-auto h-full" />
           </span>
           <strong className="font-bold">{fromToken?.symbol || 'Select'}</strong>
-          <ChevronDownIcon className="ml-px inline h-4 w-4" />
+          <ChevronDownIcon className="inline w-4 h-4 ml-px" />
         </Button>
       </div>
-      <div className="mt-1 flex items-center justify-between text-neutral-500 dark:text-neutral-400">
+      <div className="flex items-center justify-between mt-1 text-neutral-500 dark:text-neutral-400">
         <div className="mt-px text-sm">
-          {fromToken ? <NablaTokenPrice address={fromToken.id} fallback="$ -" /> : '$ -'}
+          {fromToken ? (
+            <NablaTokenPrice
+              formatByAmount={true}
+              currentTokenAmount={form.watch(fromFormFieldName)}
```

Comment:
`fromFormFieldName` is of type string. With my suggestions the type of `currentTokenAmount` should be `Big` (see my comments below).

Instead of reparsing the input string, it might be better to forward the `fromAmount` in the `Swap` component to the `From` component as the parsing is then already robustly done in `useSwapComponent`.